src/executor/attach_op_resource_pass.cc

```diff
@@ -47,8 +50,13 @@ Graph AttachOpResources(Graph g) {
     auto& requested = op_execs[nid]->op_ctx.requested;
     requested.clear();
     const auto op = inode.source->op();
-    if (fresource.count(op) != 0) {
-      auto reqs = fresource[op](inode.source->attrs);
+    const bool rsc_req = (fresource.count(op) != 0);
+    const bool rsc_ex_req = (fresource_ex.count(op) != 0);
+    if (rsc_req || rsc_ex_req) {
+      auto reqs = rsc_ex_req ? fresource_ex[op](inode.source->attrs,
+                                                dev_masks[nid],
+                                                vdispatch[nid])
+                             : fresource[op](inode.source->attrs);
```

Comment:
add CHECK(!(rsc_req && rsc_ex_req)) so that only one of them is registered

Reply:
Added.